test/xbgpu/test_engine.py

```diff
@@ -452,15 +527,34 @@ async def test_xengine_end_to_end(
         packet's timestamp (to be non-zero).
         """
         n_baselines = n_ants * (n_ants + 1) * 2
-        first_accumulation_index = 123
-        n_full_accumulations = 3
-        n_total_accumulations = n_full_accumulations + 1
+
+        # We don't want to start at 0, that's boring. So let's offset this a bit.
+        first_batch_index = 123
+
+        # The arbitrarily-selected number above might not be at a convenient
+        # place. We want to be just before a LCM of the heap accumulation
+        # indices of all the corrprods, so that we can get an incomplete
+        # accumulation as the output first.
+        lowest_common_multiple = lcm(
+            *[corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs]
+        )
+        advance = lowest_common_multiple - (first_batch_index % lowest_common_multiple)
+        first_batch_index += advance
```

Comment:
This is quite a complicated way to get an index that is a multiple of all the heap sizes. Since there is nothing special about batch 123, you could do something like
```python
first_batch_index = 123 * np.product([corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs])
```

Reply:
This was mine actually, it was part of the tweaking that I did to make sure that we were starting the test in a sane place.

I wouldn't suggest further work on this right now - your comment is valid, but rather than delaying this PR further I'd suggest just creating a small technical-debt ticket once we've got it merged.

Reply:
For the record, I've created [NGC-1048](https://skaafrica.atlassian.net/browse/NGC-1048).